README.md

```diff
@@ -33,35 +33,63 @@ To upgrade a single package named `packageName`:
 yarn upgrade packageName
 ```
 
-## Testing
-
-Tests can be run via command line with `yarn test`, from within the `/ui` directory. For more detailed reporting, use `yarn test -- --reporters=verbose`.
+## Unit Testing
+
+Unit tests can be run via command line with `yarn test`, from within the `/ui` directory. For more detailed reporting, use `yarn test -- --reporters=verbose`.
+
+
+## Local dev
+
+**cloud platform**
+1. From k8s-idpe, start the remocal service with loopback to your local ui changes.
+    1. have your remocal instance build and deployed. See [docs here](https://docs.influxdata.io/development/remocal/getting-started/).
+    1. `make remocal-dev APPS=ui DETACHED=1`
+1. From the ui directory. Build the local dev server:
+    1. set env vars for `PUBLIC` (your remocal url)
+    1. `yarn install && yarn start:dev:remocal`
+    1. `yarn link` different javascript libraries (e.g. giraffe, clockface, flux-lsp) as needed.
+        * note that the flux-lsp wasm build produces an output directory which contains a package.json for the release. Make sure to `yarn link` to this directory.
+
+**oss platform**
+1. From oss/influxdb repo, start the backend service.
+    1. setup service following the [contributing guide](https://github.com/influxdata/influxdb/blob/master/CONTRIBUTING.md#how-to-build-influxdb-from-source).
+    1. run the service, goto browser `localhost:8086`
+    1. create a login/password. (Remember these!)
+    1. **CAVEAT**: influxdb/oss does not have live reload.
+        * instead, run `make` and restart influxdb
+1. From the ui directory. Build the local dev server:
+    1. `yarn install && yarn start:dev:oss`
+    1. goto browser `localhost:8080`
+    1. the same login/password pair should work.
+    1. **CAVEAT**: live reload does not work, so you need to refresh the browser page.
 
 ## Cypress Testing
 
-e2e tests:
-For the end to end tests to run properly, the server needs to be running in the e2e testing mode with the in-memory data store.
-From the influxdb directory:
-`$ ./bin/darwin/influxd --assets-path=ui/build --e2e-testing --store=memory`
+**cloud platform**
+1. have your local dev running (see above).
+1. `export NS=<your-remocal-namespace>`
+1. run cypress test:
+    * `yarn test:e2e:remocal` to test on tsm storage
+    * `yarn test:e2e:remocal:iox` to test on iox storage
 
-**monitor-ci**
-From the ui directory. Build the javascript with
-`$ yarn start`
-To run Cypress locally with monitor-ci:
-`$ yarn cy:dev`
+**oss platform**
+1. have your local dev running (see above).
+    * Make sure to start your oss backend service with `--e2e-testing` flag.
+1. run cypress test:
+    * `yarn test:e2e:oss`
```

Comment:
This will work only once this PR is merged into the oss backend.
https://github.com/influxdata/influxdb/pull/24098

Reply:
Nice! Does that mean, once the oss backend PR is merged, we can do cypress test for oss ui locally?

Reply:
yes, exactly @appletreeisyellow !